core/src/main/java/org/apache/calcite/avatica/ColumnMetaData.java

```diff
@@ -360,6 +361,8 @@ public ColumnMetaData setRep(Rep rep) {
         builder.put(rep.clazz, rep);
       }
       builder.put(byte[].class, BYTE_STRING);
+      //The lookup code is not smart enough to handle subclasses
+      builder.put(BigDecimal.class, NUMBER);
```

Comment:
Any reason to not put this as a value for the `Rep` enum?

Reply:
Adding   `BIG_DECIMAL(BigDecimal.class, Types.NUMERIC),` works.

However, ColumnMetaData.Rep seems to be mapped to org.apache.calcite.avatica.proto.Common.Rep, so that would 
change the wire representation, which may cause compatibility issues.

I admit that I haven't traced through the whole stack, but there are multiple comments that we want to map everything to NUMBER, so it may not be desirable.

Reply:
Yeah, this is a rats nest that I am mostly responsible for. The type-wrangling is the worst part about Avatica (especially with trying to support both json and protobuf). The Javadoc on TypedValue is the best explanation I can give as to the different "representations" of data on the wire that Avatica is following.

Reply:
So I spent some more time looking at this.

AFAICT, adding a new ColumnMetaData.Rep value would not affect backwards compatibility with Protobuf, as we're already putting and expecting Common.Rep.BIG_DECIMAL on the wire anyway.

However,  with JSON we do put (the string value of) ColumnMetaData.Rep on the wire and try to parse it on the other side, so adding a new a value (using BIG_DECIMAL instead of NUMBER) would break backwards compatibility with older JSON servers, which could not parse that.

see https://calcite.apache.org/avatica/docs/json_reference.html#typedvalue

The current solution doesn't break backwards comatibility

WDYT @joshelser ?